src/main/java/com/jcabi/s3/Bucket.java

```diff
@@ -183,9 +183,16 @@ public void remove(final String key) throws IOException {
                     new Function<String, String>() {
                         @Override
                         public String apply(final String input) {
-                            return input.substring(
-                                Bucket.Prefixed.this.prefix.length()
-                            );
+                            final String result;
+                            final String head = Bucket.Prefixed.this.prefix;
+                            if ((input.length() < head.length())
+                                && (head.contains(input))
+                            ) {
+                                result = "/";
+                            } else {
+                                result = input.substring(head.length());
+                            }
+                            return result;
```

Comment:
@super132 I didn't understand this code at all... why did you change it if the ticket is about `MkBucket`?